test/resource.js

```diff
@@ -1,10 +1,159 @@
+    assert(resource.source === undefined, 'Invalid source')
+  })
+
+  it('returns the expected test data', () => {
+    const resourceDesc = {
+      'data': 'foo'
+    }
+    let resource = new Resource(resourceDesc)
+    assert(resource.source === 'foo', 'Invalid source')
+  })
+
+  it('returns the expected name', () => {
+    const resourceDesc = {
+      'name': 'FooBar'
+    }
+    let resource = new Resource(resourceDesc)
+    assert(resource.name === resourceDesc['name'], 'Invalid name')
+  })
+
+  it('recognizes that data type is local', () => {
+    const resouceDesc = {
+      'path': 'foo/bar.txt'
+    }
+    let resource = new Resource(resouceDesc)
+    assert(resource.type === 'local', 'Invalid data type')
+  })
+
+  it('recognizes that data type is remote', () => {
+    const resouceDesc = {
+      'path': 'http://www.foo.org/bar.txt'
+    }
+    let resource = new Resource(resouceDesc)
+    assert(resource.type === 'remote', 'Invalid data type')
+  })
+
+  it('recognizes that data is inline', () => {
+    const resouceDesc = {
+      'data': 'foo, bar'
+    }
+    let resource = new Resource(resouceDesc)
+    assert(resource.type === 'inline', 'Inline data not found')
+  })
+
+  it('table getter returns jts.Table', async() => {
+    const resourceDesc = {
+      'data': 'http://foofoo.org/data.csv',
+      'schema': {
+        'fields': [
+          { 'name': 'barfoo' }
+        ]
+      }
+    }
+
+    let resource = new Resource(resourceDesc)
+    let table = await resource.table
+    assert(table instanceof jts.Table,
+           'Returned object is not instance of Table')
+  })
+
+  it('table getter returns null if jsontableschama.Table throws an error',
+     async() => {
+       const resourceDesc = {
+         'data': 'http://foofoo.org/data.csv',
+       }
+       let resource = new Resource(resourceDesc)
+       let table = await resource.table
+       assert(table === null, 'Returned value not null')
+     })
+
+  describe('Tests with dp1 from data', () => {
+    let dpResources = []
+
+    beforeEach(() => {
+      _.forEach(dp1.resources, (res) => {
+        dpResources.push(res)
+      })
+    })
+
+    it('loads the resource descriptor', () => {
+      _.forEach(dpResources, (res) => {
+        let resource = new Resource(res)
+        assert(resource.descriptor === res, 'Wrong descriptor')
+      })
+    })
+
+    it('returns the correct name', () => {
+      _.forEach(dpResources, (res) => {
+        let resource = new Resource(res)
+        assert(resource.name == res.name)
+      })
+    })
+
+    it('returns the correct source', () => {
+      _.forEach(dpResources, (res) => {
+        let resource = new Resource(res)
+        assert(resource.source === res.path)
+      })
+    })
+
+    it('initialize jsontableschema.Table with csv file',
+       async(done) => {
+         let resource = new Resource({
+           "name": "dp1",
+           "format": "csv",
+           "path": "test/data/dp1/data.csv",
```

Comment:
plese move `data` to the root dir - It's a FD libs pattern to have it shared between `tests`, `examples` etc. Just for consistency between libs.